### PR TITLE
Add update_json provider

### PR DESCRIPTION
**Pull request by bobslee**
This provider will replace json and jsonb data values with a specified provider configuration per data type.

### CODE REVIEW

**Comment by bobslee** [comment]
@hkage Please can you check out this PR?
I added tests (all succeed) and tested manually (it works).
We need this feature very soon, to install from PyPi.

**Comment by hkage** [comment]
@bobslee Thanks for the contribution! I will take a look as soon as I can.

**Review by hkage — CHANGES_REQUESTED** [review]
Everything else looks good to me. And thanks again for your work and especially the tests and documentation.

File: pganonymize/providers.py [hunk 1]
```diff
@@ -246,3 +246,32 @@ class UUID4Provider(Provider):
...
+        def update_dict(input_dict, update_values_type={}):
+            """Update dictionary with recursion (nested dictionaries)."""
+            if not update_values_type:
+                return
+            for key, val in input_dict.items():
+                if isinstance(val, dict):
+                    update_dict(val, update_values_type=update_values_type)
+                else:
+                    val_type = type(val).__name__
+                    val_update = update_values_type.get(val_type)
```
[the remark this answers is not in the document]
Comment (hkage) [reply pganonymize/providers.py:267]: On the other hand in the sense of the anonymization no value would be forgotten.

File: docs/schema.rst [hunk 2]
```diff
@@ -423,6 +423,37 @@ This provider will replace values with a unique UUID4.
...
+    tables:
+      - payment_transaction:
+          fields:
+            - data:
+                provider:
+                  name: update_json
+                  update_values_type:
+                    str:
+                      provider:
+                        name: uuid
```
[the remark this answers is not in the document]
Comment (bobslee) [reply docs/schema.rst]: Done, see 29944ed

File: pganonymize/providers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bobslee) [reply pganonymize/providers.py:266]: Thanks, your're totally right! I missed this circumstance.
I added a test: `test_type_not_specified`.
It was indeed a bug and it's fixed now.

The implementation is consistent with the `pganonymize` design - which means:
Fields that are not present in the schema are not anonymized and retain their original status (value).

**Comment by bobslee** [comment]
> Everything else looks good to me. And thanks again for your work and especially the tests and documentation.

Thanks and it was my pleasure!

**Comment by hkage** [comment]
I published the new release, version 0.11.0 - thanks again for your contribution.

Don't mind the failing tests - the Python 2.7 tests don't work anymore, because the Ubuntu image dropped the binary. But the tests were successful on my workstation using a pyenv environment. As we still need Python 2.7 support I will fix this issue with the test setup in a separate pull request.

**Comment by bobslee** [comment]
You're welcome and thank you for your quick and decent follow-up.
By the way, I tested in a Python 3.8.11 env.

**Comment by bobslee** [comment]
@hkage When will the documentation be updated?
https://pganonymize.readthedocs.io/en/latest/

**Comment by hkage** [comment]
I use a webhook from Read the Docs, but it seems to be broken. Maybe because GitHub changed the permissions or the key is deactivated. I will check this.

**Comment by hkage** [comment]
The documentation is now up to date. The project was missing the now required `readthedocs.yml` and the last successful documentation build was 1 year ago, so a lot has changed on the RTD side since then. Thanks for reporting.

**Pull request merged**